iOS-DOCS.md

```diff
@@ -0,0 +1,393 @@
+
+The parameters for this method call are:
+
+* `creativeURL`: true[X] asset url returned by SSAI. In the example of Uplynk, this would correspond to `response.ads.breaks[0].ads[0].creative`
+* `adParameters`: AdParameters as returned by SSAI. In the example of Uplynk, this would correspond to `response.ads.breaks[0].ads[0].adParameters`
+* `slotType`: the type of the current ad pod, `PREROLL` or `MIDROLL`
+
+
+#### `start`
+
+```objective-c
+    - (void)start:(UIView *)baseView;
+```
+
+This method should be called by the app code when the app is ready to display the true[X] unit to the user. This can be called anytime after the unit is initialized. 
+
+The app should have as much extraneous UI hidden as possible, including player controls, status bars and soft buttons/keyboards.
+
+After calling `start`, the app code should wait for a [terminal event](#terminal-events) before taking any more action, while keeping track of whether or not the [`onAdFreePod`](#onadfreepod) event has fired.
+
+In a non-error flow, the renderer will first wait for the ad request triggered in `init` to finish if it has not already. It will then display the true[X] unit to the user in a new `View` presented on the `baseView` and then fire the [`onAdStarted`](#onadstarted) event. `onAdFreePod` and other events may fire after this point, depending on the user's choices, followed by one of the [terminal events](#terminal-events).
+
+The parameters for this method call are:
+
+* `baseView`: The `view` that `TruexAdRenderer` should present its `view` on.
+
+
+#### `stop`
+
+```objective-c
+    - (void)stop;
+```
+
+The `stop` method is only called when the app needs the renderer to immediately stop and destroy all related resources. Examples include:
+* the user backs out of the video stream to return to the normal app UI
+* there was an unrelated error that requires immediate halt of the current ad unit
+* the app code has reached a custom timeout waiting for either the [`onFetchAdComplete`](#onfetchadcomplete) or [`onAdStarted`](#onadstarted) events
+
+The `TruexAdRenderer` instance should not be used again after calling `stop` -- please remove all references to it afterwards.
+
+In contrast to `pause`, there is no way to resume the ad after `stop` is called.
+
+
+#### `pause`
+
+```objective-c
+    - (void)pause;
+```
+
+`pause` is required whenever the app needs to pause the true[X] unit (including all video, audio, and timers).
+
+
+#### `resume`
+
+```objective-c
+    - (void)resume;
+```
+
+`resume` should be called when the app is ready for a previously paused true[X] unit to resume.
+
+
+### `TruexAdRendererDelegate` Methods -- Main Flow
+
+The following `TruexAdRendererDelegate` methods signal the main flow of the `TruexAdRenderer` and may require action by the host app. The `TruexAdRendererDelegate` protocol can be found in `TruexShared.h`.
+
+
+#### `onFetchAdComplete`
```

Comment:
we say adFetchCompleted on other platforms
would onAdFetchCompleted be a better name for consistency?